src/common/darktable.c

```diff
@@ -418,51 +420,111 @@ void check_resourcelevel(const char *key, int *fractions, const int level)
   }
 }
 
-void dt_dump_pfm(
-        const char *filename,
+void _dump_pfm_file(
+        const char *name,
         const void* in,
         const int width,
         const int height,
         const dt_dump_pfm_t mode,
-        const char *modname)
+        const char *modname,
+        const int num,
+        const gboolean input,
+        const gboolean output)
 {
-  if(!darktable.dump_pfm_module) return;
-  if(!modname) return;
-  if(strcmp(modname, darktable.dump_pfm_module)) return; 
-
   if(mode > DT_DUMP_PFM_LAST)
   {
-    dt_print(DT_DEBUG_ALWAYS, "[dt_dump_pfm] undefined mode %i\n", mode); 
+    dt_print(DT_DEBUG_ALWAYS, "[dt_dump_pfm] undefined mode %i\n", mode);
+    return;
+  }
+
+  if((width < 1) || (height < 1) || (height > 20000) || (width > 20000))
+  {
+    dt_print(DT_DEBUG_ALWAYS, "[dt_dump_pfm] bad dimensions %dx%d\n", width, height);
     return;
   }
 
-  char fname[512];
-  snprintf(fname, sizeof (fname), "/tmp/darktable/%s/%s.pfm", modname, filename);
+  const size_t colors = (mode == DT_DUMP_PFM_MASK) ? 1 : 3;
+  const size_t p = (mode == DT_DUMP_PFM_MASK) ? 1 : 4;
+
+  char counts[32]= { 0 };
+  snprintf(counts, sizeof(counts), "%03d_", num);
+
+  char path[PATH_MAX]= { 0 };
+  snprintf(path, sizeof(path), "/tmp/darktable/%s", modname);
```

Comment:
Does GLib have some method for a temp folder location that is portable?

Reply:
And maybe even better :
https://docs.gtk.org/glib/func.mkdtemp.html

Reply:
In fact, should just copy what is already done for the log file, no?

Reply:
Yes, my problems were
1. Do we really want a tmporary directiory? That would be good for diskspace housekeeping but would require "us to look it up"
2. Rebooting will certainly remove that ... not a problem i guess?

I propose a darktable specific tmp directory
1. it is always created when dt is started
2. it's name will be reported in the log

Reply:
Done as discussed.

For my taste too many commits, if current code is ok for you i will squash.